packages/react-client/src/ReactFlightReplyClient.js

```diff
@@ -1151,64 +1154,86 @@ export function registerBoundServerReference<T: Function>(
     Object.defineProperties((reference: any), {
       $$FORM_ACTION: {value: $$FORM_ACTION},
       $$IS_SIGNATURE_EQUAL: {value: isSignatureEqual},
-      bind: {value: bind},
     });
+    defineBind(reference);
   }
   knownServerReferences.set(reference, {id, bound});
 }
 
+// TODO: Ideally we'd use `isServerReference` from
+// 'react-server/src/ReactFlightServerConfig', but that can only be imported
+// in a react-server environment.
+function isServerReference(reference: Object): boolean {
+  return reference.$$typeof === Symbol.for('react.server.reference');
```

Comment:
This isn't good enough to support environments that have custom server reference first class types (like Meta does and all bundlers ideally should have).

But I doubt brand checking is actually the right strategy anyway.